backend/canvas_app_explorer/canvas_lti_manager/manager.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, TypedDict
+
+from canvasapi import Canvas
+from canvasapi.tab import Tab
+
+from .data_class import ExternalToolTab
+
+class TabUpdateParams(TypedDict):
+    hidden: bool
+
+class TabAttributes(TypedDict):
+    id: str
+    course_id: int
+
+class CanvasLtiManager:
+    """
+    Adapter for canvasapi library focused on managing LTI tools in a course context
+    """
+    external_tool_prefix = 'context_external_tool_'
+
+    def __init__(self, api_url: str, api_key: str, course_id: int):
+        self.course_id = course_id
+        self.requestor = Canvas(api_url, api_key)
+
+    def create_external_tool_tab(self, tab: Tab) -> ExternalToolTab:
+        return ExternalToolTab(
+            label=tab.label,
+            id=int(tab.id.replace(self.external_tool_prefix, '')),
+            is_hidden=(hasattr(tab, 'hidden'))
+        )
+
+    def get_tools_available_in_course(self) -> List[ExternalToolTab]:
+        ex_tool_tabs = []
+        course = self.requestor.get_course(self.course_id)
+        for tab in course.get_tabs():
+            if 'external_tools' in tab.html_url:
+                ex_tool_tabs.append(self.create_external_tool_tab(tab))
+        return ex_tool_tabs
+
+    def update_tool_visibility(self, canvas_id: int, is_hidden: bool):
+        update_params: TabUpdateParams = { 'hidden': is_hidden }
+        tab_attributes: TabAttributes = {
+            'id': self.external_tool_prefix + str(canvas_id),
+            'course_id': self.course_id
+        }
+
+        tool_tab = Tab(
+            self.requestor._Canvas__requester, # Is there a better way?
```

Comment:
I don't really see a better pattern for this for now. I think it might be nice for the library to expose this as a little prettier method but I don't think what we'd do would be much different. 

Instead of storing the course as an integer we could just store the object and it has a _requester variable on it. 

I think we could also rename requestor since requester is also used to something that's clearer like canvas or canvas_api.

```
diff --git a/backend/canvas_app_explorer/canvas_lti_manager/manager.py b/backend/canvas_app_explorer/canvas_lti_manager/manager.py
index c0004ad..88efef1 100644
--- a/backend/canvas_app_explorer/canvas_lti_manager/manager.py
+++ b/backend/canvas_app_explorer/canvas_lti_manager/manager.py
@@ -19,8 +19,8 @@ class CanvasLtiManager:
     external_tool_prefix = 'context_external_tool_'
 
     def __init__(self, api_url: str, api_key: str, course_id: int):
-        self.course_id = course_id
-        self.requestor = Canvas(api_url, api_key)
+        self.canvas_api = Canvas(api_url, api_key)
+        self.course = self.canvas_api.get_course(course_id)
 
     def create_external_tool_tab(self, tab: Tab) -> ExternalToolTab:
         return ExternalToolTab(
@@ -31,8 +31,7 @@ class CanvasLtiManager:
 
     def get_tools_available_in_course(self) -> List[ExternalToolTab]:
         ex_tool_tabs = []
-        course = self.requestor.get_course(self.course_id)
-        for tab in course.get_tabs():
+        for tab in self.course.get_tabs():
             if 'external_tools' in tab.html_url:
                 ex_tool_tabs.append(self.create_external_tool_tab(tab))
         return ex_tool_tabs
@@ -41,11 +40,11 @@ class CanvasLtiManager:
         update_params: TabUpdateParams = { 'hidden': is_hidden }
         tab_attributes: TabAttributes = {
             'id': self.external_tool_prefix + str(canvas_id),
-            'course_id': self.course_id
+            'course_id': self.course.id
         }
 
         tool_tab = Tab(
-            self.requestor._Canvas__requester, # Is there a better way?
+            self.course._requester,
             tab_attributes
         )
         data = tool_tab.update(**update_params)
```

Reply:
Okay, @jonespm, I renamed the class variable to be `canvas_api`.

I was about to make the change you suggested about attaching the `course` as a variable as well, but then I realized that what you're suggesting would mean we'd call `/courses/xx` with `GET` every time we created a new `CanvasLtiManager` instance (when finding the tools and also whenever it's updated). That seemed like it could be a performance drag, so I'm opting to keep things as they are for now.

This also led me to realize that we may not need to call `canvas_api.get_course` in `get_tools_available_in_course` either, since we could probably use a similar pattern to what we're doing with `Tab` and just pass a requester and the course ID to `Course`. I didn't want to mess with this too much right now though, and perhaps there's an advantage to keeping the syntax idiomatic and clean in this method. WDYT? We can open a separate issue if you think it's important. The one extra call here with the initial tool load doesn't seem like as big a deal as repeated course calls every time a tab is updated.